FirebaseFunctionsSwift/Sources/FunctionsSerializer.swift

```diff
@@ -172,47 +166,24 @@ class FUNSerializer: NSObject {
     }
   }
 
-  private func decodeWrappedType(_ wrapped: NSDictionary) throws -> AnyObject? {
-    let type = wrapped["@type"] as! String
-
-    guard let value = wrapped["value"] as? String else {
-      return nil
-    }
-
+  private func decodeWrappedType(_ type: String, _ value: String) throws -> AnyObject? {
     switch type {
     case Constants.longType:
       let formatter = NumberFormatter()
       guard let n = formatter.number(from: value) else {
-        // TODO: Throw FUNInvalidNumberError(value, wrapped);
-        return nil
+        throw SerializerError.invalidValueForType(value: value, requestedType: type)
       }
-
       return n
     case Constants.unsignedLongType:
       // NSNumber formatter doesn't handle unsigned long long, so we have to parse it.
-      // let str = value.utf8
-      // TODO: Port this atrocity
-      throw SerializerError.unimplemented
-    /*
-     const char *str = value.UTF8String;
-     char *end = NULL;
-     unsigned long long n = strtoull(str, &end, 10);
-     if (errno == ERANGE) {
-       // This number was actually too big for an unsigned long long.
-       if (error != NULL) {
-         *error = FUNInvalidNumberError(value, wrapped);
-       }
-       return nil;
-     }
-     if (*end) {
-       // The whole string wasn't parsed.
-       if (error != NULL) {
-         *error = FUNInvalidNumberError(value, wrapped);
-       }
-       return nil;
-     }
-     return @(n);
-     */
+      let str = (value as NSString).utf8String
+      var endPtr: UnsafeMutablePointer<CChar>?
+      let returnValue = UInt64(strtoul(str, &endPtr, 10))
```

Comment:
Ah, thanks for porting this. I saw `stroul` but was looking for the `unsigned long long` variant, but now I realize that we're only expecting an unsigned long. Thanks!